src/Parameters/RestrictiveChecker.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Neomerx\JsonApi\Parameters;
+
+use \Neomerx\JsonApi\Contracts\Codec\CodecMatcherInterface;
+use Neomerx\JsonApi\Contracts\Parameters\CombinedCheckerInterface;
+use \Neomerx\JsonApi\Contracts\Parameters\ParametersInterface;
+use \Neomerx\JsonApi\Contracts\Integration\ExceptionThrowerInterface;
+
+/**
+ * @package Neomerx\JsonApi
+ */
+class RestrictiveChecker implements CombinedCheckerInterface
+{
+
+    /**
+     * @var RestrictiveHeaderChecker
+     */
+    private $headerChecker;
+
+    /**
+     * @var RestrictiveParameterChecker
+     */
+    private $parameterChecker;
+
+    /**
+     * @param ExceptionThrowerInterface $exceptionThrower
+     * @param CodecMatcherInterface     $codecMatcher
+     * @param bool                      $allowUnrecognized
+     * @param array|null                $includePaths
+     * @param array|null                $fieldSetTypes
+     * @param array|null                $sortParameters
+     * @param array|null                $pagingParameters
+     * @param array|null                $filteringParameters
+     */
+    public function __construct(
+        ExceptionThrowerInterface $exceptionThrower,
+        CodecMatcherInterface $codecMatcher,
+        $allowUnrecognized = false,
+        array $includePaths = null,
+        array $fieldSetTypes = null,
+        array $sortParameters = null,
+        array $pagingParameters = null,
+        array $filteringParameters = null
+    ) {
+        $this->headerChecker = new RestrictiveHeaderChecker(
```

Comment:
Please don't create them internally with `new` it makes testing more difficult and makes it less modular. You can pass `Interfaces` instead and add method to `Factory` that creates it with default implementations.